src/lib/hooks/useZAuctionAPI.ts

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+
+//the hook that returns the useZauctionAPI with the api endpoint that match with the selected chain
+
+export function useZAuctionAPI(chain: number): string {
+	const [api, setApi] = useState('Unsupported Chain');
```

Comment:
Instead of "Unsupported Chain" this should be `undefined`.

Therefore it would be `useState<string | undefined>` where a string means a valid endpoint, but undefined is an invalid endpoint (unsupported chain)

You will then have to check `api` to ensure that is a `string` (and not undefined) before using it.